crates/types/src/scenarios.rs

```diff
@@ -0,0 +1,370 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use solana_clock::Slot;
+use uuid::Uuid;
+
+use crate::Idl;
+
+// ========================================
+// Core Scenarios Types
+// ========================================
+
+/// Defines how an account address should be determined
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum AccountAddress {
+    /// A specific public key
+    Pubkey(String),
+    /// A Program Derived Address with seeds
+    Pda {
+        program_id: String,
+        seeds: Vec<PdaSeed>,
+    },
+}
+
+/// Seeds used for PDA derivation
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum PdaSeed {
+    String(String),
+    Bytes(Vec<u8>),
+    Pubkey(String),
+    /// Reference to a property value
+    PropertyRef(String),
+}
+
+/// A reusable template for creating account overrides
+/// Values are mapped directly to IDL fields using dot notation (e.g., "agg.price", "expo")
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OverrideTemplate {
+    /// Unique identifier for the template
+    pub id: String,
+    /// Human-readable name
+    pub name: String,
+    /// Description of what this template does
+    pub description: String,
+    /// Protocol this template is for (e.g., "Pyth", "Switchboard")
+    pub protocol: String,
+    /// IDL for the account structure - defines all available fields and types
+    pub idl: Idl,
+    /// How to determine the account address
+    pub address: AccountAddress,
```

Comment:
This `AccountAddress` type seems unnecessary to me - what am I missing? I'd think it could just be address.

The protocol with this IDL will be deployed to an address, and the accounts it generates will have their PDA derivations defined in the IDL.

I'm not sure of protocols themselves being deployed to PDAs.